main.tf

```diff
@@ -45,6 +45,7 @@ resource "ibm_iam_authorization_policy" "policy" {
 
 # workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
 resource "time_sleep" "wait_for_authorization_policy" {
+  count      = var.kms_encryption_enabled == false || var.skip_iam_authorization_policy ? 0 : 1
```

Comment:
if this now has a count, does the depends_on on line 56 need to be updated?

Reply:
Instead of duplication line 38 and 48, suggest to use a local (incase the logic is ever updated, it would only need to be changed in 1 place)

Reply:
> if this now has a count, does the depends_on on line 56 need to be updated?

No, when doing depends on we don't do any indexing or anything like that so it passes as is and is still necessary